File: backend/src/main/java/com/civicworks/model/redis/SocialPost.java
package com.civicworks.model.redis;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.index.Indexed;
import java.util.List;

@RedisHash("SocialPost")
@Data
public class SocialPost {
    @Id private String id;
    @Indexed private Long projectId;
    
    private String content;
    private String authorName;
    private Long authorId;
    private long timestamp;
    
    @Indexed private String channel; 
    
    // Proposal Data
    private Long budgetProposalId;
    private String budgetAmount; 
    
    // Rich Media
    private List<String> imageUrls; 
    private String documentUrl;
    private String audioUrl;
    
    private MediaType mediaType;

    public enum MediaType { TEXT, IMAGE, DOCUMENT, AUDIO }
}

File: frontend/lib/models/social_post.dart
class SocialPost {
  final String? id;
  final String authorName;
  final String content;
  final int timestamp;
  final String channel; 
  final List<String> imageUrls;
  
  // Budget Props
  final int? budgetProposalId;
  final String? budgetAmount;
  
  // Rich Media
  final String? documentUrl;
  final String? audioUrl;
  final String? mediaType; // IMAGE, DOCUMENT, AUDIO, TEXT

  SocialPost({
    this.id, required this.authorName, required this.content,
    required this.timestamp, required this.channel,
    this.imageUrls = const [],
    this.budgetProposalId, this.budgetAmount,
    this.documentUrl, this.audioUrl, this.mediaType
  });

  factory SocialPost.fromJson(Map<String, dynamic> json) {
    return SocialPost(
      id: json['id'],
      authorName: json['authorName'] ?? 'Anon',
      content: json['content'] ?? '',
      timestamp: json['timestamp'] ?? 0,
      channel: json['channel'] ?? 'PUBLIC',
      imageUrls: (json['imageUrls'] as List<dynamic>?)?.map((e) => e.toString()).toList() ?? [],
      budgetProposalId: json['budgetProposalId'],
      budgetAmount: json['budgetAmount'],
      documentUrl: json['documentUrl'],
      audioUrl: json['audioUrl'],
      mediaType: json['mediaType'] ?? 'TEXT',
    );
  }
}
